feature_tests/js/api/BorrowedFields.mjs

```diff
@@ -36,25 +36,19 @@ export class BorrowedFields {
     // of arrays for each lifetime to do so. It accepts multiple lists per lifetime in case the caller needs to tie a lifetime to multiple
     // output arrays. Null is equivalent to an empty list: this lifetime is not being borrowed from.
     _intoFFI(
-        slice_cleanup_callbacks,
+        functionCleanup,
         appendArrayMap
     ) {
-        slice_cleanup_callbacks.push((appendArrayMap[aAppendArray] || []).length > 0 ? () => { for (let lifetime of appendArrayMap[aAppendArray]) { appendArrayMap[aAppendArray].push(a); } a.garbageCollect(); } : a.free);
-        
-        slice_cleanup_callbacks.push((appendArrayMap[aAppendArray] || []).length > 0 ? () => { for (let lifetime of appendArrayMap[aAppendArray]) { appendArrayMap[aAppendArray].push(b); } b.garbageCollect(); } : b.free);
-        
-        slice_cleanup_callbacks.push((appendArrayMap[aAppendArray] || []).length > 0 ? () => { for (let lifetime of appendArrayMap[aAppendArray]) { appendArrayMap[aAppendArray].push(c); } c.garbageCollect(); } : c.free);
-        
-        return [diplomatRuntime.DiplomatBuf.str16(wasm, this.#a), diplomatRuntime.DiplomatBuf.str8(wasm, this.#b), diplomatRuntime.DiplomatBuf.str8(wasm, this.#c)]
+        return [(appendArrayMap["aAppendArray"].length > 0 ? diplomatRuntime.CleanupArena.createWith(appendArrayMap["aAppendArray"]) : functionCleanup).alloc(diplomatRuntime.DiplomatBuf.str16(wasm, this.#a)), (appendArrayMap["aAppendArray"].length > 0 ? diplomatRuntime.CleanupArena.createWith(appendArrayMap["aAppendArray"]) : functionCleanup).alloc(diplomatRuntime.DiplomatBuf.str8(wasm, this.#b)), (appendArrayMap["aAppendArray"].length > 0 ? diplomatRuntime.CleanupArena.createWith(appendArrayMap["aAppendArray"]) : functionCleanup).alloc(diplomatRuntime.DiplomatBuf.str8(wasm, this.#c))]
```

Comment:
could you explain what is going on here?

Reply:
It's meant to mirror Dart's implementation. It's deciding whether or not to create a `CleanupArena` with a given lifetime (`aAppendArray`) or the function's `CleanupArena` (which is freed as soon as the function returns). The `alloc` function returns whatever is passed into it, while also adding that item to everything that will be cleaned up once the function is done executing. So this is a way to get JS friendly struct info while also ensuring that slices will be cleaned up when appropriate (either right after the function, or using `createWith` which uses a FinalizationRegistry).

Reply:
ah, i see. worth calling it functionCleanupArena or something?

Reply:
Yeah, I'll rename it that for clarity